src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataReaderTest/DataReaderTest.cs

```diff
@@ -329,6 +330,207 @@ public static void CheckLegacyNullRowVersionIsEmptyArray()
+                }
+            }
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static async void ExecuteReader_CommandBehavior_SequentialAccess_Async_Test6()
+        {
+            using SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString);
+            connection.Open();
+
+            using var cmd = connection.CreateCommand();
+            cmd.CommandText = "SELECT 'abcdefg', 'hijklmn'";
+            using var reader = cmd.ExecuteReader(CommandBehavior.SequentialAccess);
+            {
+                reader.Read();
+                using var tr1 = reader.GetTextReader(0);
+                {
+                    reader.GetValue(1);
+                    char[] value = new char[1];
+                    await Assert.ThrowsAsync<ObjectDisposedException>(() => tr1.ReadAsync(value, 0, 1));
+                }
+            }
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static void ExecuteReader_CommandBehavior_SequentialAccess_Test7()
+        {
+            using SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString);
+            connection.Open();
+
+            using var cmd = connection.CreateCommand();
+            cmd.CommandText = "SELECT 'abcdefg', 'hijklmn'";
+            using var reader = cmd.ExecuteReader(CommandBehavior.SequentialAccess);
+            {
+                reader.Read();
+                using var tr1 = reader.GetTextReader(0);
+                using var tr2 = reader.GetTextReader(1);
+
+                tr2.Read();
+                Assert.Throws<ObjectDisposedException>(() => tr1.Read());
+            }
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static async void ExecuteReader_CommandBehavior_SequentialAccess_Async_Test8()
+        {
+            using SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString);
+            connection.Open();
+
+            using var cmd = connection.CreateCommand();
+            cmd.CommandText = "SELECT 'abcdefg', 'hijklmn'";
+            using var reader = cmd.ExecuteReader(CommandBehavior.SequentialAccess);
+            {
+                reader.Read();
+                using var tr1 = reader.GetTextReader(0);
+                using var tr2 = reader.GetTextReader(1);
+
+                char[] value2 = new char[1];
+                await tr2.ReadAsync(value2, 0, 1);
+                char[] value1 = new char[1];
+                await Assert.ThrowsAsync<ObjectDisposedException>(() => tr1.ReadAsync(value1, 0, 1));
+            }
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static void ExecuteReader_CommandBehavior_SequentialAccess_Test9()
```

Comment:
Better Method Name

Reply:
Method name renamed to one that conforms to the test naming convention:

The name of your test should consist of three parts:

The name of the method being tested.
The scenario under which it's being tested.
The expected behavior when the scenario is invoked.